scripts/logging.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+#
+# Helper functions for console (stdout/stderr) logging.
+#
+
+function get_datetime_str() {
+    date +"%D %T %Z"
+}
+
+# Emits argument in $1 to stdout with date-time and "INFO" as header.
+function log() {
+    printf "[%s] INFO : %s\n" "$(get_datetime_str)" "$1"
+}
+
+# Emits argument in $1 to stdout+stderr with date-time and "ERROR" as header.
+function log_err() {
+    printf "[%s] ERROR: %s\n" "$(get_datetime_str)" "$1" 1>&2
+}
```

Comment:
* Full timestamp is better.
* Command date could failed, added error handling.
* Added green, yellow and red colors.
* Prints all parameters allows to write `log My name is "$USER".`.

I've been testing similar functions for years, see [`log` in `base.sh`](https://github.com/rdavid/shellbase/blob/master/lib/base.sh#L269).

Reply:
It is very necessary. `$(timestamp)` runs in sub-shell, so `exit` of `timestamp` exits from the sub-shell. `log` will not exit. You can run functions directly in shell, in this case no need for double exit:
```
#/bin/sh
timestamp() {
  exit 1
}
timestamp
```

Reply:
Ah, thx, I forgot about the subshell exection. What do you think about this version of the timestamp func?:
```
timestamp() {
    printf "[%s]" "$(date --utc +"%D-%T" 2>/dev/null || printf "")"
}
```
If `date` fails, its output is silently discarded (sent to /dev/null) and the output of `timestamp()` will be just an empty string `"[]"`. It's shorter, no local vars needed, and it won't break code execution and the log trace will be always printed. With your version, if `date` fails, the whole program crashes and the log trace isn't even seen...

Reply:
There is an interesting approach.

>It's shorter
We run the `timestamp` in a new process. Effectiveness is not the point. And your approach adds another new process, which is an opposite of shorter.

>no local vars needed
The var exists only because I want to keep the `date` error message under control.

> won't break code execution and the log trace will be always printed
`date` is a very basic system call, like `printf`, if it fails something goes wrong with the Linux kernel configuration, I believe in the concept 'to fail earlier'. A job result is not reliable, if `date` fails. Stop and change the kernel configuration or something. In order to debug an issue you have to have the error message.

The `date` format doesn't work on Mac:
```
~ » date --utc +"%D-%T"
date: illegal option -- -
usage: date [-jnRu] [-I[date|hours|minutes|seconds]] [-f input_fmt]
            [-r filename|seconds] [-v[+|-]val[y|m|w|d|H|M|S]]
            [[[[mm]dd]HH]MM[[cc]yy][.SS] | new_date] [+output_fmt]
```

Reply:
Classic POSIX Unix commands do not support long flags, everything that starts with `--some-long-flag`.

Reply:
@rdavid thanks for your suggestions. I've updated the logging funcs to use your code, with some minor changes.

When I said that my timestamp func version was "shorter", I was refering to readability only, which is my (very personal) primary goal when coding, and especially when writing bash, for obvious reasons.